src/Features/Core/Portable/Navigation/DefaultSymbolNavigationService.cs

```diff
@@ -5,13 +5,18 @@
 #nullable disable
 
 using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.MetadataAsSource;
 using Microsoft.CodeAnalysis.Options;
 
 namespace Microsoft.CodeAnalysis.Navigation
 {
     internal class DefaultSymbolNavigationService : ISymbolNavigationService
     {
+        public Task<MetadataAsSourceFile> GetAndOpenGeneratedFileAsync(ISymbol symbol, Project project, CancellationToken cancellationToken)
+            => Task.FromResult<MetadataAsSourceFile>(null);
```

Comment:
It seems a bit odd to me that we're adding this interface -- the presumption here is this is something being called behind the scenes of TryNavigateToSymbol. Is the only need that we need to know the file path that we ended up with? Should we update TryNavigateToSymbol's signature to expose that instead?

Reply:
TryNavigateToSymbol does some extra work that I think is unnecessary in the LSP go-to-def metadata scenario, so I didn't think we would want to use it here. In the LSP go-to-def metadata scenario, all we want to do is get the file and open the document.